app/graphql/theme-setup.tsx

```diff
@@ -0,0 +1,40 @@
+import { useThemeMode } from "@rneui/themed"
+import React from "react"
+import { Appearance } from "react-native"
+import { useDarkModeQuery } from "./generated"
+
+export const ThemeSync = () => {
+  const { mode, setMode } = useThemeMode()
+  const { data } = useDarkModeQuery()
```

Comment:
```suggestion
  const isDarkMode = useDarkMode()
  
  useEffect(()=> {
  const intendedMode = isDarkMode ? "dark" : "light"
  if (intendedMode !== mode) {
      setMode(intendedMode)
  }
  }, [isDarkMode, mode])
```
Could this not replace the entire file and be functionally the same?